dynamiq/memory/backend/qdrant.py

```diff
@@ -21,122 +22,167 @@ class Qdrant(MemoryBackend):
 
     name = "Qdrant"
 
-    def __init__(self, connection: QdrantConnection, embedder: BaseEmbedder, index_name: str = "conversations"):
-        """Initializes the Qdrant memory storage."""
+    def __init__(
+        self,
+        connection: QdrantConnection,
+        embedder: BaseEmbedder,
+        index_name: str = "conversations",
+        metric: str = "cosine",
+        on_disk: bool = False,
+    ):
+        """Initializes the Qdrant memory storage.
+
+        Args:
+            connection: QdrantConnection instance for connecting to Qdrant
+            embedder: Embedder instance for creating embeddings
+            index_name: Name of the collection to store messages
+        """
         self.connection = connection
         self.index_name = index_name
         self.embedder = embedder
 
         try:
-            self.client = self.connection.connect()
+            self.vector_store = QdrantVectorStore(
+                connection=connection,
+                index_name=index_name,
+                dimension=embedder.dimensions,
+                create_if_not_exist=True,
+                metric=metric,
+                on_disk=on_disk,
+                recreate_index=False,
+            )
+            self.client = self.vector_store._client
+            if not self.client:
+                raise QdrantError("Failed to initialize Qdrant client")
+            # Ensure collection is properly set up
+            self.vector_store._set_up_collection(
+                collection_name=index_name,
+                embedding_dim=embedder.dimensions,
+                create_if_not_exist=True,
+                recreate_collection=False,
+                similarity=metric,
+                use_sparse_embeddings=False,
+                sparse_idf=False,
+                on_disk=on_disk,
+            )
```

Comment:
Do we really need to duplicate this code? 
I was thinking that the collection setup would be executed during init or actual client usage with parameters from the init

Reply:
Updated

Reply:
Please update it in the next PR!)